FBSimulatorControl/Interactions/FBSimulatorInteraction.m

```diff
@@ -91,6 +91,26 @@ - (instancetype)bootSimulator
       return [[[FBSimulatorError describe:@"Could not obtain process info for booted simulator process"] inSimulator:simulator] failBool:error];
     }
 
+    // Waitng for all required processes to start
+    NSSet *requiredProcesses = [NSSet setWithArray:
+    @[
+      @"SpringBoard",
+      @"com.apple.accessibility.AccessibilityUIServer",
+      @"com.apple.audio.SystemSoundServer-iOS-Simulator",
+      @"AssetCacheLocatorService",
+      @"MobileCal",
+      @"medialibraryd",
+    ]];
+    BOOL didStartAllRequiredProcesses = [[NSRunLoop mainRunLoop] spinRunLoopWithTimeout:60 untilTrue:^BOOL{
+      NSSet *runningProcesses = [NSSet setWithArray:[simulator.processQuery processesWithLaunchPathSubstring:nil]];
```

Comment:
You don't have to get all processes there is `subprocessesOf:` query that you can use on the `launchd_sim` process identifier that you have from the `FBSimulatorLaunchInfo`.

In fact you could end up querying for  (and finding) these subprocesses of a *different* simulator.